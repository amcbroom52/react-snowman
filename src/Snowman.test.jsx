import { render, fireEvent, screen } from "@testing-library/react";
import {expect, it, test} from "vitest";
import Snowman from "./Snowman";

it("Doesn't show buttons once player loses", function () {
  const {container} = render(<Snowman words={["test"]}/>);

  const aButton = container.querySelector("[value = a]");
  const bButton = container.querySelector("[value = b]");
  const cButton = container.querySelector("[value = c]");
  const dButton = container.querySelector("[value = d]");
  const fButton = container.querySelector("[value = f]");
  const gButton = container.querySelector("[value = g]");

  fireEvent.click(aButton);
  fireEvent.click(bButton);
  fireEvent.click(cButton);
  fireEvent.click(dButton);
  fireEvent.click(fButton);
  fireEvent.click(gButton);

  const loseElement = screen.getByText("You Lose!");

  expect(aButton).not.toBeInTheDocument();
  expect(loseElement).toBeInTheDocument();
});

it("Matches losing snapshot", function () {
  const {container} = render(<Snowman words={["test"]}/>);

  const aButton = container.querySelector("[value = a]");
  const bButton = container.querySelector("[value = b]");
  const cButton = container.querySelector("[value = c]");
  const dButton = container.querySelector("[value = d]");
  const fButton = container.querySelector("[value = f]");
  const gButton = container.querySelector("[value = g]");

  fireEvent.click(aButton);
  fireEvent.click(bButton);
  fireEvent.click(cButton);
  fireEvent.click(dButton);
  fireEvent.click(fButton);
  fireEvent.click(gButton);

  expect(container).toMatchSnapshot();
})
