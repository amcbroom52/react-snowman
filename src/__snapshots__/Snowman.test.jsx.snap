// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches losing snapshot 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="/src/6.png"
    />
    <p>
      Wrong guesses: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
    </p>
    <p>
      You Lose!
    </p>
  </div>
</div>
`;
